activejob/lib/active_job/core.rb

```diff
@@ -137,6 +141,11 @@ def deserialize(job_data)
       self.exception_executions = job_data["exception_executions"]
       self.locale               = job_data["locale"] || I18n.locale.to_s
       self.timezone             = job_data["timezone"] || Time.zone.try(:name)
+      self.enqueued_at          = job_data["enqueued_at"]
+    end
+
+    def enqueded_for
+      Time.now - enqueued_at
```

Comment:
Do we actually need to calculate how long the job has been in the queue?  I would expect the log messages to have a timestamp.  If the log message has a time stamp, we can just log the "enqueued time" and whatever reads the log messages can calculate how long the message sat in the queue.  Logging just the duration means that we lose the "enqueued_at" information (or we have to take the log timestamp and subtract the duration to back in to the value).

Reply:
In our application we only lock the enqueued_at.